compiler/sem/sempass2.nim

```diff
@@ -1684,8 +1682,7 @@ proc trackProc*(c: PContext; s: PSym, body: PNode) =
   if cap != nil:
     # the procedure captures something and thus requires a hidden environment
     # parameter
-    if s.kind == skMacro or
-       (isCompileTimeProc2(s) and not isCompileTimeProc2(cap.sym.owner)):
+    if not canCaptureFrom(s, cap.sym.owner):
```

Comment:
I _believe_ it's the same logic if we drop the negation and swap the body of this `if` branch with that of the `else`.

I prefer avoiding negations, but by no means a hard requirement.

Reply:
Not quite, the explicit calling convention check has to happen before setting the convention. The `elif` branch could of course be moved into the `if` branch as a nested `if` statement, but I think doing so would make the logic harder to follow.

Reply:
D'oh, didn't pick-up on that, thanks for clarifying.